File: solutions/ruby/acronym/1/acronym.rb
module Acronym

  def self.abbreviate expansion
    result = ''

    for word in expansion
      .tr('-', ' ')
      .tr('_', '')
      .tr("'", '')
      .tr(',', '')
      .split

      result << word.capitalize.chr
    end

    result
  end

end

File: solutions/elixir/acronym/1/lib/acronym.ex
defmodule Acronym do

  # Regex splits on whitespace,
  # dashes and underscores
  @delimiters ~r/[\s_-]+/

  def custom_split(str) when is_binary(str) do
    String.split(str, @delimiters)
  end

  def abbreviate(phrase) do
    phrase
    |> custom_split()
    |> Enum.map(&String.first(&1))
    |> Enum.join()
    |> String.upcase()
  end

end
